src/main/java/io/snowdrop/reporting/weekreport/WeeklyDevelopmentReportImpl.java

```diff
@@ -57,37 +60,37 @@ public static WeeklyDevelopmentReportImpl build(final Date pStartDate, final Dat
    *
    * @return
    */
-  public String buildWeeklyReport() {
-    StringBuilder sb = new StringBuilder();
+  public String buildWeeklyReport(String reportName) {
+    StringBuilder sb = new StringBuilder(MarkdownHelper.addHeadingTitle(reportName, 1)).append(ReportConstants.CR);
     ZonedDateTime now = ZonedDateTime.now();
     ZonedDateTime twoWeeksAgo = now.minusWeeks(2);
     ZonedDateTime oneMonthAgo = now.minusMonths(1);
     String repoName = ReportConstants.WEEK_DEV_REPO_OWNER + "/" + ReportConstants.WEEK_DEV_REPO_NAME;
     Issue.findByIssuesForWeeklyDevelopmentReport(repoName, startDate, endDate).list().stream()
-        .collect(Collectors.groupingBy(Issue::getAssignee, Collectors.toSet()))
-        .entrySet().forEach(eachAssignee -> {
+    .collect(Collectors.groupingBy(Issue::getAssignee, TreeMap::new, Collectors.toSet()))
+    .entrySet().forEach(eachAssignee -> {
       String assignee = eachAssignee.getKey();
-      sb.append(ReportConstants.CR).append(ReportConstants.CR).append(MarkdownHelper.addHeadingTitle(assignee, 2)).append(ReportConstants.CR);
+      String associateName = Associate.getAssociateName(assignee, IssueSource.GITHUB);
+      sb.append(ReportConstants.CR).append(ReportConstants.CR).append(MarkdownHelper.addHeadingTitle(associateName, 2)).append(ReportConstants.CR);
       UnorderedList labelUnorderedList = new UnorderedList();
-      eachAssignee.getValue().stream().collect(Collectors.groupingBy(Issue::getLabel, Collectors.toSet())).entrySet().forEach(eachLabel -> {
+      eachAssignee.getValue().stream().collect(Collectors.groupingBy(Issue::getLabel, TreeMap::new, Collectors.toSet())).entrySet().forEach(eachLabel -> {
         String label = eachLabel.getKey();
         labelUnorderedList.getItems().add(label);
         UnorderedList issueUnorderedList = new UnorderedList();
         eachLabel.getValue().stream().forEach(eachIssue -> {
           TextBuilder issueTextB = new TextBuilder();
           Date dateCreatedAt = eachIssue.getCreatedAt();
-          String strMdColor = "gray";
-          if(eachIssue.isOpen()) {
+          String strMdColor = " ![#a9a9a9](https://via.placeholder.com/15/a9a9a9/000000?text=+) ";
```

Comment:
Can we please externalize those in application.properties?

Reply:
Done.